subsys/net/lib/zperf/zperf_common.c

```diff
@@ -237,7 +237,9 @@ static int zperf_init(void)
 	zperf_udp_uploader_init();
 	zperf_tcp_uploader_init();
 
-	zperf_session_init();
+	if (IS_ENABLED(CONFIG_NET_ZPERF_SERVER)) {
```

Comment:
FYI, my zperf concurrent session support in #88239 will conflict with this one as it needs session support also in client mode. Let's see which PR gets merged first, the other one will then need adjustments.